crates/shared/src/bad_token/trace_call.rs

```diff
@@ -113,48 +123,56 @@ impl TraceCallDetector {
         requests
     }
 
-    fn handle_response(traces: &[BlockTrace], amount: U256) -> Result<TokenQuality> {
+    fn handle_response(
+        traces: &[BlockTrace],
+        amount: U256,
+        take_from: H160,
+    ) -> Result<TokenQuality> {
         ensure!(traces.len() == 8, "unexpected number of traces");
 
         let gas_in = match ensure_transaction_ok_and_get_gas(&traces[1])? {
             Ok(gas) => gas,
             Err(reason) => {
                 return Ok(TokenQuality::bad(format!(
-                    "can't transfer into settlement contract: {reason}"
+                    "Transfer of token from on chain source {take_from:?} into settlement \
+                     contract failed: {reason}"
                 )))
             }
         };
+        let arbitrary = Self::arbitrary_recipient();
         let gas_out = match ensure_transaction_ok_and_get_gas(&traces[4])? {
             Ok(gas) => gas,
             Err(reason) => {
                 return Ok(TokenQuality::bad(format!(
-                    "can't transfer out of settlement contract: {reason}"
+                    "Transfer token out of settlement contract to arbitrary recipient \
+                     {arbitrary:?} failed: {reason}",
                 )))
             }
         };
 
+        let message = "Failed to decode the token's balanceOf response because it did not return 32 bytes. A common cause of this is a bug in the Vyper smart contract compiler. See https://github.com/cowprotocol/services/pull/781 for more information.";
```

Comment:
Could this line be wrapped?

Reply:
It seems like cargo fmt sometimes autowraps long strings and sometimes doesn't. I'm hesitant to manually wrap this because it means every time we change the text we need to manually reflow a couple of lines. I'd rather have one long line. But no strong feeling.

This is how I would manually format it:
```rust
        let message = "\
            Failed to decode the token's balanceOf response because it did not \
            return 32 bytes. A common cause of this is a bug in the Vyper \
            smart contract compiler. See \
            https://github.com/cowprotocol/services/pull/781 for more \
            information.\
        ";
```

Reply:
I think reflowing the text should not be a problem (I don't expect it to happen often). But I also don't have a strong opinion on this. Feel free to ignore.

Reply:
Not wrapping causes `cargo fmt` to sometimes ignore formatting for long lines. I do think we should should manually wrap until `cargo fmt` can take over.

Reply:
Stable rustfmt has never wrapped lines for you (right?). Nightly sometimes does. With this message, regardless of whether I manually wrap it or don't wrap it, nightly rustfmt does not handle it.